### PR TITLE
Add example usage of "extract_fields" for mlflow.search_traces in documentation

### PR DESCRIPTION
**Pull request by xq-yin**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/xq-yin/mlflow/pull/12319?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12319/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12319
```

</p>
</details>

### What changes are proposed in this pull request?

Currently we don't have examples of how to use "extract_fields" for mlflow.search_traces in our doc: https://mlflow.org/docs/latest/llms/tracing/index.html#searching-for-traces
In fact, this argument can be super important to help developers build evaluation datasets from traces. This PR introduces a simple example to our documentation to promote the awareness/usage.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests

```
cd docs
make clean livehtml
```
image

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [X] Yes. I've updated:
  - [X] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [X] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: docs/source/llms/tracing/index.rst [hunk 1]
```diff
@@ -467,6 +467,17 @@ filter strings, and other parameters.
...
+Alternatively, you can use `fluent API <./index.html#tracing-fluent-apis>`_ to search for traces. This method allows you to specify fields to extract from traces using the format 
+``"span_name.[inputs|outputs]"`` or ``"span_name.[inputs|outputs].field_name"``.
+
+.. code-block:: python
+
+    # Search for traces in specific experiments
+    traces = mlflow.search_traces(
+        experiment_ids=["1", "2"],
+        extract_fields=["span1.inputs", "span1.outputs.c"],
+    )
```
Comment (dbczumar) [inline docs/source/llms/tracing/index.rst:479]: Can we show an example output? E.g. using a screenshot?

**Review by dbczumar — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/12319/files#r1635628963 is addressed. Thanks @xq-yin !

File: docs/source/llms/tracing/index.rst (hunk 1 above)
Comment (WeichenXu123) [reply docs/source/llms/tracing/index.rst:479]: do we need a screenshot?
we can just copy output text of `print(traces)`

File: docs/source/llms/tracing/index.rst [hunk 2]
```diff
@@ -467,6 +467,17 @@ filter strings, and other parameters.
         experiment_ids=["1", "2"], filter_string="attributes.status = 'OK'", max_results=5
     )
 
+Alternatively, you can use `fluent API <./index.html#tracing-fluent-apis>`_ to search for traces. This method allows you to specify fields to extract from traces using the format 
+``"span_name.[inputs|outputs]"`` or ``"span_name.[inputs|outputs].field_name"``.
```
Comment (dbczumar) [inline docs/source/llms/tracing/index.rst:471]: Can we also explain:

1. The extracted fields are included as extra columns in the pandas dataframe returned by `search_traces())
2. This feature can be used to build evaluation datasets to further improve model / agent performance

File: docs/source/llms/tracing/index.rst [hunk 3]
```diff
@@ -467,6 +467,17 @@ filter strings, and other parameters.
         experiment_ids=["1", "2"], filter_string="attributes.status = 'OK'", max_results=5
     )
 
+Alternatively, you can use `fluent API <./index.html#tracing-fluent-apis>`_ to search for traces. This method allows you to specify fields to extract from traces using the format
```
Comment (B-Step62) [inline docs/source/llms/tracing/index.rst:470]:
```suggestion
Alternatively, you can use fluent API :py:func:`mlflow.search_traces` to search for traces, which returns a pandas DataFrame with each row contains a trace. This method allows you to specify fields to extract from traces using the format
```

For easier navigation to the API doc where we have more examples.

**Pull request merged**